src/AbstractModuleLocator.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Dhii\Modular\Locator;
+
+/**
+ * Common functionality for module locators.
+ *
+ * @since [*next-version*]
+ */
+abstract class AbstractModuleLocator
+{
+    /**
+     * Parameter-less constructor.
+     * 
+     * Invoke this in the real constructor.
+     * 
+     * @since [*next-version*]
+     */
+    protected function _construct()
+    {
+    }
+
+    /**
+     * Reads a config source into a standard config.
+     * 
+     * @since [*next-version*]
+     * 
+     * @return array The standard config.
+     */
+    abstract protected function _read($config);
+
+    /**
+     * Retrieves configuration sources.
+     *
+     * @since [*next-version*]
+     *
+     * @return array Configuration sources.
+     */
+    abstract protected function _getSources();
+
+    /**
+     * Locates module configurations.
+     *
+     * @since [*next-version*]
+     *
+     * @return mixed[] A list of module configurations.
+     */
+    protected function _locate()
+    {
+        $configs = [];
+        foreach ($this->_getSources() as $_source) {
+            $configs[md5($_source)] = $this->_read($_source);
```

Comment:
The hashing done here should be in a separate method. I assume that this is what `$this->_generateKeyFromSource()` was created for. Perhaps you forgot to change this line?